Realm/RLMSyncSessionRefreshHandle.mm

```diff
@@ -53,20 +53,38 @@ - (instancetype)initWithFullURLPath:(NSString *)urlPath
 }
 
 - (void)invalidate {
-    [self.timer invalidate];
-    self.user = nil;
+    dispatch_async(dispatch_get_main_queue(), ^{
+        [self.timer invalidate];
+        self.user = nil;
+    });
 }
 
 - (void)scheduleRefreshTimer:(NSTimeInterval)fireTime {
-    static const NSInteger refreshBuffer = 10;
-    [self.timer invalidate];
-    self.timer = [[NSTimer alloc] initWithFireDate:[NSDate dateWithTimeIntervalSince1970:(fireTime - refreshBuffer)]
-                                          interval:0
-                                            target:self
-                                          selector:@selector(timerFired:)
-                                          userInfo:nil
-                                           repeats:NO];
-    [[NSRunLoop currentRunLoop] addTimer:self.timer forMode:NSDefaultRunLoopMode];
+    constexpr NSInteger REFRESH_BUFFER = 10;
```

Comment:
This naming style is typically reserved for macros. Shouldn't this be a `NSTimeInterval`?